src/extension_commands.ml

```diff
@@ -158,6 +158,48 @@ let _open_ocamllsp_output_pane, _open_ocaml_platform_ext_pane, _open_ocaml_comma
       (handler Output.command_output_channel) )
 ;;
 
+let decorationType =
+  let options =
+    DecorationRenderOptions.create
+      ~backgroundColor:(`ThemeColor (ThemeColor.make ~id:"editorHoverWidget.foreground"))
+      ~color:(`ThemeColor (ThemeColor.make ~id:"editorHoverWidget.background"))
+      ~border:"1px solid"
+      ~borderColor:(`ThemeColor (ThemeColor.make ~id:"editorHoverWidget.border"))
+      ~isWholeLine:false
+      ()
+  in
+  Window.createTextEditorDecorationType ~options
+;;
+
+let set_decoration text_editor range =
+  let _decorationOptions =
+    let renderOptions =
+      let before = ThemableDecorationAttachmentRenderOptions.create () in
+      let options = ThemableDecorationInstanceRenderOptions.create ~before () in
+      Some (DecorationInstanceRenderOptions.create ~light:options ~dark:options ())
+    in
+    DecorationOptions.create ~range ~renderOptions ()
+  in
+  TextEditor.setDecorations
+    text_editor
+    ~decorationType
+    ~rangesOrOptions:(`Ranges [ range ])
+;;
+
+let remove_decoration text_editor =
+  let rangesOrOptions = `Ranges [] in
+  TextEditor.setDecorations text_editor ~decorationType ~rangesOrOptions
+;;
```

Comment:
I think you should move the decoration related code to a sub-module `Decorations` and rename `set_decoration` to a more specific name like `highlight_range` and `remove...` to `remove_all_highlightings`.